### DISCUSSION TITLE
Vega-Lite conditional colour based on values

### DISCUSSION THREAD

**Discussion by naveenwills** [vega/vega-lite#8814] [Q&A] [answered]
Hello all,

I'm very new to the language and exploring Vega-lite to implement a solution at work. I'm trying to create a calendar heatmap using Deneb in power bi, based on percentage value for each day. 
I want the heatmap colour to show based on conditions. 
0 = blue
1-20 = Orange 
21-70 = Brown
71- 100 = Red

The code I have is as below
"color": {
      "field": "DailyOccupancy %",
      "type": "quantitative",
      "legend": {
        "title": "Occupancy sensor data",
        "orient": "top-left", 
        "type": "gradient",
        "cornerRadius": 2, 
        "direction": "horizontal",
        "offset": -90
      },
      "scale": {
        "scheme": {
          "name": "blueorange"
        }
      }
    }

![image](https://user-images.githubusercontent.com/129381573/228839172-65fbebcb-1a51-4289-bb0b-cef12dc84240.png)

I'm not able to figure out a way to add conditions to this code. Please any help would be appreciated.

**Comment by mattijn** [comment, marked as answer]
You can make use of a `threshold` including `domain` and `range` for the `scale` definition:
```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
  "data": {"url": "data/cars.json"},
  "mark": "rect",
  "encoding": {
    "y": {"field": "Origin", "type": "nominal"},
    "x": {"field": "Cylinders", "type": "ordinal"},
    "color": {
      "aggregate": "mean",
      "field": "Horsepower",
      "scale": {
        "type": "threshold",
        "domain": [80, 110, 140],
        "range": ["blue", "orange", "brown", "red"]
      },
      "type": "quantitative",
      "legend":{"orient":"top"}
    }
  }
}
```
![image](https://user-images.githubusercontent.com/5186265/228848221-d641ad4c-7382-41f0-9a35-9d11a7d3efd9.png)

[Open the Chart in the Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEySYUqUAwBOgtCrVICUJNohSZ8gL5Lk2gNb7tcKJkUg4sqG2U1ZZNFAAT0CQADMaOEFlfQB5bRpaeSVMIJw4fVk2BD8kPRsQAA9QiKiY9ABhIOFZZTgzd1T0-TZtX1k8kALvQVbQpDIyZwpMDPQEOCRk8Mjo-QAJVog4HDYAd3r3aDyx0CaxkHpnCBg2OaVlbKQ-NABtAA4ABgUARhfnl4AWR4BdJW0pmQxrcQExBAwMkpWoDIaDtOtps4Yj8Cvt9ABHBhTTB0NQ0UjuQTkTzlUCtSKyNzoTBsHBdKwMoA)

**Reply by naveenwills** [reply]
Exactly what I wanted, thanks a lot :)
